contrib/src/main/java/com/datatorrent/contrib/kafka/KafkaTopicOffsetManager.java

```diff
@@ -0,0 +1,487 @@
+      int partitionCount)
+  {
+
+    partitionList = new LinkedList<TopicAndPartition>();
+
+    offsetsMap = new HashMap<TopicAndPartition, Long>();
+
+    setHostname(hostname);
+    setConsumerGroup(consumerGroup);
+    setClientId(clientID);
+    setTopic(topic);
+    setPartitionCount(partitionCount);
+    setPort(port);
+    setRetryCount(retryCount);
+
+    getConnectionHandle(getRetry());
+  }
+
+  public void getConnectionHandle(int retry)
+  {
+
+    if (retry == 0) {
+      setChannel(null);
+      return;
+    }
+
+    int correlationID = 0;
+    try {
+      BlockingChannel channel = new BlockingChannel(getHostname(), getPort(), BlockingChannel.UseDefaultBufferSize(),
+          BlockingChannel.UseDefaultBufferSize(), (int)getWaitTimeout() * 10);
+      channel.connect();
+
+      //TODO - do a get until you get no further data from the zk/topic
+
+      for (int i = 0; i < getPartitionCount(); i++) {
+        final TopicAndPartition tempPartition = new TopicAndPartition(getTopic(), i);
+        partitionList.add(tempPartition);
+
+      }
+
+      channel.send(new ConsumerMetadataRequest(consumerGroup, (short)1, correlationID++, getClientId()));
+      ConsumerMetadataResponse metadataResponse = ConsumerMetadataResponse.readFrom(channel.receive().buffer());
+
+      // Set the incremental correlation ID
+      setCorrelationId(correlationID);
+
+      if (metadataResponse.errorCode() == ErrorMapping.NoError()) {
+        Broker offsetManager = metadataResponse.coordinator();
+        // if the coordinator is different, from the above channel's
+        // host then reconnect
+        channel.disconnect();
+        channel = new BlockingChannel(offsetManager.host(), offsetManager.port(),
+            BlockingChannel.UseDefaultBufferSize(), BlockingChannel.UseDefaultBufferSize(), (int)getWaitTimeout() * 10);
+
+        channel.connect();
+        setChannel(channel);
+        LOG.info("Channel initialized");
+
+      } else {
+        setRetry(getRetry() - 1);
+        getConnectionHandle(getRetry());
+      }
+    } catch (Exception e) {
+      LOG.error("Not able to initialize the channel");
+    }
+
+  }
+
+  /**
+   * @return the retry
+   */
+  public int getRetry()
+  {
+    return retry;
+  }
+
+  /**
+   * @param retry
+   *          the retry to set
+   */
+  public void setRetry(int retry)
+  {
+    this.retry = retry;
+  }
+
+  /**
+   * @return the waitTimeout
+   */
+  public long getWaitTimeout()
+  {
+    return waitTimeout;
+  }
+
+  /**
+   * @param waitTimeout
+   *          the waitTimeout to set
+   */
+  public void setWaitTimeout(long waitTimeout)
+  {
+    this.waitTimeout = waitTimeout;
+  }
+
+  public void fetchOffsets()
+  {
+    // version 1 and above fetch from Kafka, version 0 fetches from
+    // ZooKeeper
+
+    OffsetFetchRequest fetchRequest = new OffsetFetchRequest(getConsumerGroup(), getPartitionList(), (short)1,
```

Comment:
Please document what the parameter 1 means or better, make it a property and document it there.

https://cwiki.apache.org/confluence/display/KAFKA/Committing+and+fetching+consumer+offsets+in+Kafka

(short) 1 /* version */); // version 1 and above commit to Kafka, version 0 commits to ZooKeeper